### DISCUSSION TITLE
How to Print Array of Objects passed as paramters to method??

### DISCUSSION THREAD

**Discussion by constantmanish** [btraceio/btrace#606] [Q&A]
i have a method method1(List<Expression>)
How would i be able to print "value" attribute present in Expression class by tracing this method1?

I am not allowed to use loop , neither i can use get() function, is there any way i can get away with this problem??

**Comment by constantmanish** [comment]
@jbachorik , can you Please help me on this??

**Reply by jbachorik** [reply]
Hi @constantmanish - unfortunately, this is not supported OOTB.
Can you use the `@Trusted` mode which will allow you to call methods directly on the `List`?

**Reply by constantmanish** [reply]
HI @jbachorik , 

Thank you so much for reply. trusted mode helped me for my use case. will we be able to trace java.util.* class as well in trusted mode ??

**Reply by jbachorik** [reply]
Yes. In trusted mode the BTrace script becomes just a regular Java code. It is up to you then to think of the possible consequences of performing certain operations - eg. is iterating over a huge collection done in the context of a traced function cause latency spikes or health checks to fail etc.

**Reply by constantmanish** [reply]
Thank you so much for your response @jbachorik , your response has helped me alot in figuring out some deep rooted bugs in our system. One last question , is there any workaround that would help me to trace java.util classes? I think they are ignored while tracing, any workaround in trusted mode would be of great help if there is any.

**Reply by jbachorik** [reply]
Sorry for the late reply - unfortunately, not as of now.
I have a long-running branch with some improved linking (https://github.com/btraceio/btrace/tree/jb/linking_1) which should help here but the changes are very complex and I still haven't ironed out all wrinkles there :(
